### PR TITLE
Add installation request for missing dependencies (bunq/tinker_php#18)

### LINKED ISSUE #18: Add installation request for required dependencies
When a dependency is detected as missing it should ask the user if the dependency should be installed with the default installation command instead of simply exiting the setup script.

For composer the `php-mbstring `and `php-curl` packages are also required which are currently not checked yet

### PR DESCRIPTION
**Pull request by Crecket**
## This PR closes/fixes the following issues:
 - Closes bunq/tinker_php#18
    - [x] Tested

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -129,7 +145,17 @@ function determineInstructionInstallation
     systemName="${2}"
     programPackageName=${3:-${programName}}
     commandInstallationConstantName="${COMMAND_INSTALLATION_PREFIX}$(capitalize ${systemName})"
-    printf "${ERROR_COULD_NOT_FIND_COMMAND}" "${programName}" "${!commandInstallationConstantName} ${programPackageName}"
+
+    warningMessage=$(printf "${ERROR_COULD_NOT_FIND_COMMAND}" "${programName}" "${!commandInstallationConstantName} ${programPackageName}")
+
+    echo ${warningMessage} >$(tty)
+    echo "Do you want to try and install this now? (y/N)? " >$(tty)
```
Comment (OGKevin) [inline setup.sh]: Please move this string into a consant.

File: setup.sh [hunk 2]
```diff
@@ -57,7 +57,23 @@ function assertAllPrerequisitePresent
...
+            assertAllPrerequisitePresent true
+        fi
+    fi
+}
+
+function assertComposerPrerequisitePresent
+{
+    phpModules=$(php -m | head)
+
+    if [[ ${phpModules} != *"curl"* ]]; then
```
Comment (OGKevin) [inline setup.sh]: There are some extension missing according to: [composer.json](https://github.com/bunq/sdk_php/blob/8c1faefc111d9b970168a1837ca165d854954941/composer.json)

Aslo this seems like code repeation. Please create a new method:

1. isModlueMissing(string[] $allModules, string $moduleToFind): bool
 You can do this via:

```
allModule=$(php -m)
allNeededModule=(some smart way to get a string[] of everthig in composer json)


for line in $allNeededModule; do
  if (isModlueMissing($allModule $line)); then
      determineInstructionInstallation ...
done
```

Your method `assertComposerPrerequisitePresent` will then loop over each needed and preform the check. 

To get all the needed extensions, i would suggest to use the latest composer json on master branch and then run a regex for everything with `ext-`

**Review by OGKevin — CHANGES_REQUESTED** [review]
Some minor refactoring please

File: setup.sh [hunk 3]
```diff
@@ -5,6 +5,8 @@ ERROR_SYSTEM_UNKNOWN='Unknown system found "%s".\n'
 ERROR_RUN_IN_EMPTY_DIRECTORY='Please run the script from an empty directory\n'
 ERROR_COULD_NOT_FIND_COMMAND='Could not find "%s", try installing it by running "%s".'
 
+MESSAGE_INSTALL_DEPENDENCY='Do you want to try and install this now? (y/N)? '
```
Comment (einfachdavid89) [inline setup.sh:8]: Y

**Review by einfachdavid89 — APPROVED** [review]

File: setup.sh (hunk 3 above)
Comment (einfachdavid89) [inline setup.sh:8]: Y

**Review by einfachdavid89 — APPROVED** [review]

**Review by einfachdavid89 — APPROVED** [review]

**Pull request closed without merging**